AutoMapper.AspNetCore.OData.EFCore/LinqExtensions.cs

```diff
@@ -123,23 +123,17 @@ public static Expression<Func<T, bool>> ToFilterExpression<T>(this ODataQueryOpt
         /// <returns></returns>
         public static Expression<Func<IQueryable<T>, IQueryable<T>>> GetQueryableExpression<T>(this ODataQueryOptions<T> options, ODataSettings oDataSettings = null)
         {
-            if (NoQueryableMethod(options, oDataSettings))
```

Comment:
Because we don't want to return null on `NoQueryableMethod`  we always need to append some orderBy: https://github.com/AutoMapper/AutoMapper.Extensions.OData/issues/221

Reply:
I don't agree.  `Microsoft.AspNetCore.OData.Query.EnableQueryAttribute` does not sort by default either which leads me think that the logic enforcing the "ThenBy PK" sort really belongs in your own code.  We should close this.

Reply:
Before closing this, just want to point out few things:

- Please notice there are 2 parts here:
     1. Applying `OrderBy` even when it was not requested at all. 
     2. When one passes some `OrderBy` we append `ThenBy` PK as well
- Saying that "Microsoft doesn't do it so we shouldn't do it" isn't really a valid argument...
- It is difficult to do it after `GetQueryAsync` since EF apply the `orderBy` after the LIMIT which ruins the point:
```
SELECT *
FROM (
    SELECT *
    FROM "Cases" AS "c"
    ORDER BY "c"."Type" DESC
    LIMIT @__p_0
) AS "t"
ORDER BY "t"."Id"
```
- This is a general//real issue during paging, solving this can help us make paging consistent, it can also opens us to `Keyset Pagination` https://google.aip.dev/158

Maybe we can settle for 2. ("When one passes some `OrderBy` we append `ThenBy` PK as well")

Reply:
It's the other way round.  Changing the code needs justification - leaving it untouched doesn't. 

Adding the `ThenBy` to your request when necessary is better.  Then it does not have to run for users who are not requesting it.

Reply:
Done, added `ThenBy` only when requested

Reply:
I think you misunderstood.  That logic [already exists](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/a19b58a1ab5980223eef7aa3b50f25ea974296ab/AutoMapper.AspNetCore.OData.EFCore/LinqExtensions.cs#L214).   The [tests](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/a19b58a1ab5980223eef7aa3b50f25ea974296ab/AutoMapper.OData.EFCore.Tests/GetQueryTests.cs#L503) should confirm that.

I'm going to close this one.  We can reopen if I'm missing something.

Reply:
Correct - just do this: `"/corebuilding?$top=5&$expand=Builder($expand=City),Tenant&$orderby=Name desc,Identity"; ` Otherwise we'd be adding the `ThenBy` for every user on every request.

Reply:
IMO adding thenBy pk on every request is best practice.

I don't want the consistency of my paging depending on customers (whether they pass Identity or not) also notice I don't change the request itself, it is immutable, internally im taking care of adding thenBy...

Reply:
It should work if we add an opt-in flag to [ODataSettings](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/a19b58a1ab5980223eef7aa3b50f25ea974296ab/AutoMapper.AspNetCore.OData.EFCore/ODataSettings.cs#L10).  Something like `ODataSettings.AlwaysSortByPrimaryKey` then we can do both behaviors when that flag is set (PK sort when `NoQueryableMethod `plus the `ThenBy` PK) .

Reply:
Done, thanks!